python/pyarrow/tests/test_table.py

```diff
@@ -1376,6 +1376,28 @@ def test_concat_tables_with_different_schema_metadata():
     assert table2.schema.equals(table3.schema)
 
 
+def test_concat_tables_with_promote_option():
+    t1 = pa.Table.from_arrays(
+        [pa.array([1, 2], type=pa.int64())], ["int64_field"])
+    t2 = pa.Table.from_arrays(
+        [pa.array([1.0, 2.0], type=pa.float32())], ["float_field"])
+
+    result = pa.concat_tables([t1, t2], promote=True)
```

Comment:
```suggestion
    with pytest.warns(FutureWarning):
        result = pa.concat_tables([t1, t2], promote=True)
```

This asserts the warning is raised and at the same time also ensures we don't unnecessarily see the warning in the pytest logs